crates/sqlexec/src/remote/client.rs

```diff
@@ -0,0 +1,257 @@
+use crate::errors::{ExecError, Result};
+use protogen::gen::rpcsrv::service::{
+    execution_service_client::ExecutionServiceClient, CloseSessionRequest, CloseSessionResponse,
+    ExecuteRequest, ExecuteResponse, InitializeSessionRequest, InitializeSessionResponse,
+};
+use proxyutil::metadata_constants::{
+    COMPUTE_ENGINE_KEY, DB_NAME_KEY, ORG_KEY, PASSWORD_KEY, USER_KEY,
+};
+use std::sync::Arc;
+use tonic::{
+    metadata::MetadataMap,
+    transport::{Channel, Endpoint},
+    IntoRequest, Response, Status, Streaming,
+};
+use url::Url;
+
+const DEFAULT_RPC_PROXY_PORT: u16 = 6443;
+
+/// Params that need to be set on grpc connections when going through the proxy.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ProxyAuthParams {
+    /// User (generated Cloud credentials)
+    pub user: String,
+    /// Password (generated Cloud credentials)
+    pub password: String,
+    /// DB name
+    pub db_name: String,
+    /// Org name.
+    pub org: String,
```

Comment:
Is this allowed to be name or id, or always just name?

Reply:
`org` can be name or id. We have logic to check that and send the appropriate params: https://github.com/GlareDB/glaredb/blob/main/crates/proxyutil/src/cloudauth.rs#L106-L124

`db_name` can only be name.

I'd like for us to phase out being able to provide org ID (https://github.com/GlareDB/glaredb/blob/main/crates/proxyutil/src/cloudauth.rs#L58-L60). As a user, I'm only ever going to use my org name. I think ID is sticking around for the frontend stuff.

Reply:
Currently it is used for FE, yes. Should make those changes in Cloud (+verify), which would allow us to pull that impl out from here as "dead code" afterwards.

Reply:
https://github.com/GlareDB/cloud/issues/1960